kerchunk/utils.py

```diff
@@ -116,9 +140,32 @@ def rename_target_files(
         ujson.dump(new, f)
 
 
-def _encode_for_JSON(store):
+def zarr_init_group_and_store(store=None, zarr_version=None, overwrite=True):
+    zarr_version = zarr_version or 2
```

Comment:
Can you clarify the semantics of `zarr_version` here (and in `zarr_open`)? What's the behavior of each of these cases?

| zarr-python-library-version | zarr_version | behavior |
| --- | --- | --- |
| 2.x | None | write zarr v2 |
| 2.x | 2 | write zarr v2 |
| 2.x | 3 | error |
| 3.x | None | write zarr v2 or v3? |
| 3.x | 2 | write zarr v2 |
| 3.x | 3 | write zarr v3 |

Really it's just the case of `zarr_version=None` with zarr-python 3.x that I'm unsure about. i.e. what's the default behavior: write zarr v2, or write whatever version is the default for that version of zarr-python?

Reply:
Also worth confirming that we *do* error for `zarr_version=3` with zarr-python=2, to not silently ignore that keyword argument.

Reply:
Surprisingly, if you have zarrv2 installed and you set `zarr_version=3` then `zarr` _will_ accept that, issue a warning, and give you a valid group. I need to re-run these tests with zarrv3 installed and see what happens in those cases.

Reply:
Zarr2 (i.e., mainline released version) has had v3 internally for quite a while, but the implementation is different. It should still conform to the v3 spec, though!